### DISCUSSION TITLE
Swagger docs doesn't show 401 responses when using `Depends` on  a `SecurityBase`-descended class

### DISCUSSION THREAD

**Discussion by kruton** [fastapi/fastapi#12116] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# From the tutorial at:
# https://fastapi.tiangolo.com/tutorial/security/first-steps/#fastapis-oauth2passwordbearer

from typing import Annotated

from fastapi import Depends, FastAPI
from fastapi.security import OAuth2PasswordBearer

app = FastAPI()

oauth2_scheme = OAuth2PasswordBearer(tokenUrl="token")


@app.get("/items/")
async def read_items(token: Annotated[str, Depends(oauth2_scheme)]):
    return {"token": token}
```

### Description

Following the Security First Steps tutorial doesn't produce the `openapi.json` or Swagger UI output it indicates in the tutorial. This causes automatic OpenAPI generation tools to omit the possibility of a response code of 401 authorization failure.

Steps to reproduce:
* Creating a minimal FastAPI example that matches the security tutorial and save as `main.py`
* Run "`fastapi dev main.py`"
* Open Swagger UI at `http://localhost:8000/docs`
* Compare the output to [What It Does](https://fastapi.tiangolo.com/tutorial/security/first-steps/#what-it-does) from the tutorial
* Observe that the Swagger UI docs don't show the possibility of "401 Unauthorized" response

### Operating System

Linux, macOS

### FastAPI Version

0.112.2

### Pydantic Version

2.8.2

### Python Version

Python 3.12.4

### Additional Context

The generated Swagger UI docs are listed below. It doesn't show the possibility of 401 response code. I have tried various other ways to inherit from `SecurityBase` but none seem to automatically include the response as the tutorial documentation indicates.

<img width="751" alt="Screenshot 2024-09-02 at 12 05 22 PM" src="https://github.com/user-attachments/assets/9c0d4806-32ce-4cab-8f35-0c74bff7d07d">

**Comment by YuriiMotov** [comment, marked as answer]
Actually, the documentation does not state that FastAPI will add this response model automatically, does it?

To get the [picture that you can see in the mentioned article](https://fastapi.tiangolo.com/tutorial/security/first-steps/#what-it-does) you need to make an unauthorized request to that endpoint.

But I think it would be really nice if FastAPI could add this response automatically for protected endpoints.

**Reply by kruton** [reply]
Oh, yes, you're correct. I missed that it's under the "Server Response" area. I read the part above as indicating FastAPI's automatic generation is where it came from:

> All the security utilities that integrate with OpenAPI (and the automatic API docs) inherit from SecurityBase, that's how FastAPI can know how to integrate them in OpenAPI.

**Comment by davidzenisu** [comment]
Bottom line is it's not supported at the moment, is my understanding correct?
@kruton have you considered opening an issue to request this feature/do you think it would make sense to open one?

**Reply by kruton** [reply]
I think it would be useful given how many libraries can generate bindings from OpenAPI `.json` files. For instance the TypeScript generator [openapi-typescript](https://www.npmjs.com/package/openapi-typescript) won't expect a 401 error if it's not told it's a possibility and the static analysis of that code won't show the error handling failure.

I think it's fine to just insert 401 errors for Security annotations, but maybe having the dependency have an annotation that can be read would also suffice to satisfy this use case.
